crates/holochain/src/core/ribosome/guest_callback/validate.rs

```diff
@@ -309,51 +312,45 @@ mod slow_tests {
 
     #[tokio::test(flavor = "multi_thread")]
     async fn test_validate_implemented_invalid_return() {
-        let validate_invocation = ValidateInvocation::new(
-            ZomesToInvoke::One(IntegrityZome::from(TestWasm::ValidateInvalidReturn).erase_type()),
-            &Op::RegisterAgentActivity(RegisterAgentActivity {
-                action: SignedActionHashed::new_unchecked(
-                    Action::CreateLink(fixt!(CreateLink)),
-                    fixt!(Signature),
-                ),
-                cached_entry: None,
-            }),
+        let unit_entry_def = EntryDef::default_from_id("unit");
+        let zome = InlineZomeSet::new_unique_single(
+            "integrity",
+            "coordinator",
+            vec![unit_entry_def.clone()],
+            0,
         )
-        .unwrap();
-
-        let ribosome = RealRibosomeFixturator::new(Zomes(vec![TestWasm::ValidateInvalidReturn]))
-            .next()
-            .unwrap();
-
-        let err = ribosome
-            .run_validate(fixt!(ValidateHostAccess), validate_invocation)
-            .await
-            .unwrap_err();
-
-        let_assert!(RibosomeError::CallbackInvalidReturnType(err_msg) = err);
-        assert!(err_msg.starts_with("invalid value: integer `42`"));
-    }
+        .function("integrity", "validate", |_api, _op: Op| Ok(42usize))
+        .function("coordinator", "create", move |api, ()| {
+            let entry = Entry::app(().try_into().unwrap()).unwrap();
+            let hash = api.create(CreateInput::new(
+                InlineZomeSet::get_entry_location(&api, EntryDefIndex(0)),
+                EntryVisibility::Public,
+                entry,
+                ChainTopOrdering::default(),
+            ))?;
+            Ok(hash)
+        });
```

Comment:
You can omit the coordinator and `.call_fallible::<_, ()>(&cell.zome("integrity"), "init", ())` further down.
```suggestion
        .function("integrity", "validate", |_api, _op: Op| Ok(42usize));
```